lint: try disable t.Parallel()

otan opened the pull request:
Output for lint with t.Parallel() is very confusing. This PR attempts to
remove some parallels and balance that with time taken on CI.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/51155)
<!-- Reviewable:end -->

otan committed:
lint: try disable t.Parallel()

Output for lint with t.Parallel() is very confusing. This PR attempts to
remove some parallels and balance that with time taken on CI.

Release note: None

otan commented:
Lint took 29mins without Parallel(), which is about the same as other runs (https://teamcity.cockroachdb.com/viewType.html?buildTypeId=Cockroach_UnitTests_Check&branch_Cockroach_UnitTests=%3Cdefault%3E&tab=buildTypeStatusDiv)

I think this change should go in.

otan commented:
bors r=tbg
